processing/src/main/java/org/apache/carbondata/processing/loading/csvinput/CSVInputFormat.java

```diff
@@ -281,6 +281,7 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext context)
           Charset.forName(CarbonCommonConstants.DEFAULT_CHARSET));
 
       CsvParserSettings settings = extractCsvParserSettings(job);
+      settings.setReadInputOnSeparateThread(false);
```

Comment:
This will hit data loading performance, it think better to expose one property with default value true, if user don't want in in different thread he can set this to false. 
I think this change is not related to Complex type please raise in separate PR